packages/analytics-js/src/services/ErrorHandler/ErrorHandler.ts

```diff
@@ -57,37 +95,64 @@ class ErrorHandler implements IErrorHandler {
     }
   }
 
-  onError(error: SDKError, context = '', customMessage = '', shouldAlwaysThrow = false) {
+  onError(
+    error: SDKError,
+    context = '',
+    customMessage = '',
+    shouldAlwaysThrow = false,
+    errorType = 'handled',
+  ) {
     // Error handling is already implemented in processError method
     let errorMessage = processError(error);
 
     // If no error message after we normalize, then we swallow/ignore the errors
     if (!errorMessage) {
       return;
     }
-
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const errorState: ErrorState = {
+      severity: 'error',
+      unhandled: errorType !== 'handled',
+      severityReason: { type: errorType },
+    };
     errorMessage = removeDoubleSpaces(
       `${context}${LOG_CONTEXT_SEPARATOR}${customMessage} ${errorMessage}`,
     );
 
     let normalizedError = error;
     // Enhance error message
     if (isTypeOfError(error)) {
-      (normalizedError as Error).message = errorMessage;
+      try {
+        (normalizedError as Error).message = errorMessage;
+      } catch (e) {
+        // ignore if message property can not be updated
+      }
     } else {
       normalizedError = new Error(errorMessage);
     }
+    if (errorType === 'handled') {
+      // TODO: Remove the below line once the new Reporting plugin is ready
+      this.notifyError(normalizedError as Error);
 
-    this.notifyError(normalizedError as Error);
-
-    if (this.logger) {
-      this.logger.error(errorMessage);
+      if (this.logger) {
+        this.logger.error(errorMessage);
 
-      if (shouldAlwaysThrow) {
+        if (shouldAlwaysThrow) {
+          throw normalizedError;
+        }
+      } else {
         throw normalizedError;
       }
+    }
+
+    if (
+      state.reporting.isErrorReportingEnabled.value &&
+      !state.reporting.isErrorReportingPluginLoaded.value
+    ) {
+      // buffer the error
+      this.errorBuffer.enqueue([error, errorState]);
```

Comment:
Instead of an array, why don't we use an object type with fields like error instance and state? It's less ambiguous.

Reply:
Addressed